TheWarWithin/RogueAssassination.lua

```diff
@@ -2096,24 +2101,39 @@ spec:RegisterAbilities( {
         cooldown = 0,
         gcd = "totem",
         school = "physical",
-
         spend = 35,
         spendType = "energy",
 
         startsCombat = true,
         cycle = function () return buff.deadly_poison.up and "deadly_poison_dot" or buff.amplifying_poison.up and "amplifying_poison_dot" or nil end,
 
+        cp_gain = function()
+            local fanCP = buff.clear_the_witnesses.up and 2 or 1
+            
+            -- Predict crit gains
+            if talent.seal_fate.enabled then
+                -- calculate the crit chance of Fan of Knives then estimate 
+                fanCP = fanCP + floor( true_active_enemies * ( 0.01 * ( crit_pct_current + ( talent.deadly_precision.enabled and 5 or 0 ) + ( talent.thrown_precision.enabled and 5 or 0 ) + ( buff.momentum_of_despair.up and 10 or 0 ) + ( buff.master_assassin_any.up and 20 or 0 ) ) ) )
```

Comment:
Just noting that this deviates from normal design of not asserting that something happens if we can't confirm it will happen, but we'll use this as a test case and see how it plays out.  Technically, the emulated game state is *more* likely to be correct when calculating based on crits with multiple enemies, but we may provide a follow-up recommendation that can't actually be used if you don't proc all the expected combo points.

Reply:
So far I've had 3 users test and it's been noted by all that it's a big improvement.

Pre-change: Addon assumes in 5 targets you need about 5 fan of knives to reach the CP threshold.
Post-change: Addon assumes in 5 targets you need about 2.

So instead of reacting from 5 casts down to 1 on lucky crits, you go from 2 > 1 which just feels like any other proc. **_More_** accurate, not completely accurate. Performance wise, it stops the addon from making it to the very bottom of the list for every single recommendation because it thinks it can only get 1 CP.

I know it's not perfect, but it does provide a significantly better user experience. Ideally they just stop designing stuff like this. :) I can keep an eye on feedback about it, too. If it's often happening that you can't quite use the finisher yet, I can always decrement the overall crit gains by 1 for a safety net.

"naked" opener, no CDs, no stealth, etc. Using my estimated Combo point gains. Suggest 2, it's pretty much always either 1 or 2 outside of tiny tiny chance to crit nothing twice in a row. On this test, I procced and it only took 1 fan so rupture moved up a slot.
![image](https://github.com/user-attachments/assets/d8c9714e-e77e-4522-b247-fb638a52d862)

Identical situation using the current file on your published branch. On this test it took exactly 2 fans like my prediction would have said.
![image](https://github.com/user-attachments/assets/f9b8cfc2-8afb-4ca5-ab7d-2bba5c304312)

Reply:
> Just noting that this deviates from normal design of not asserting that something happens if we can't confirm it will happen, but we'll use this as a test case and see how it plays out.  Technically, the emulated game state is *more* likely to be correct when calculating based on crits with multiple enemies, but we may provide a follow-up recommendation that can't actually be used if you don't proc all the expected combo points.

The estimate is still conservative (sure to using `floor`), but more likely to be an accurate estimate of a future game state, so I think it's an improvement over the extremely conservative estimate for CP gain from FoK before the changes.

Reply:
I think the main worry isn't the 3rd or 4th recommendation in the queue, but the 2nd recommendation. If the 1st recommendation is to FoK on 3 targets and we predict 5 CP but get only 3, then perhaps the 2nd recommendation is a finisher that suddenly converts to another FoK after the initial recommendation is cast. I could certainly see that as a jarring user experience.

Reply:
Yeah, even that is way better than the current state. And like I said if it starts happening often enough we can always decrement by 1 to smooth it out for sure, no issue there. But you already get pretty high crit fast anyways due to talents. 61% crit on FoK at like 606 ilevel, no stat optimization.

Reply:
There is precedent for turning this into a user preference -- Warriors had a mechanic in Shadowlands that would work on crit and I had a slider that let you set how much crit you needed for it to be considered a "sure thing."

Also, the underlying goal of finding ways to short-circuit the priority faster is a good one.  I've been putting a lot of thought into modifications into changes that can stop sooner.  We can talk about that elsewhere.